core/src/main/java/com/valtech/aem/saas/core/fulltextsearch/DefaultFulltextSearchService.java

```diff
@@ -50,71 +57,85 @@ public int getRowsMaxLimit() {
+      boolean enableAutoSuggest,
+      boolean enableBestBets) {
+    SaasIndexValidator.getInstance().validate(index);
     String requestUrl = getRequestUrl(index, fulltextSearchGetRequestPayload);
     log.debug("Search GET Request: {}", requestUrl);
     Optional<SearchResponse> searchResponse = searchRequestExecutorService.execute(new SearchRequestGet(requestUrl));
     if (searchResponse.isPresent()) {
       printResponseHeaderInLog(searchResponse.get());
-      return getFulltextSearchResults(searchResponse.get());
+      return getFulltextSearchResults(searchResponse.get(), enableAutoSuggest, enableBestBets);
     }
     return Optional.empty();
   }
 
-  private void validateIndex(String index) {
-    if (StringUtils.isBlank(index)) {
-      throw new IllegalArgumentException(
-          "SaaS index name is missing. Please configure index name in Context Aware configuration.");
-    }
+  private String getRequestUrl(String index, FulltextSearchGetRequestPayload fulltextSearchGetRequestPayload) {
+    return String.format("%s%s",
+        getApiUrl(index),
+        fulltextSearchGetRequestPayload.getPayload());
   }
 
-  private Optional<FulltextSearchResults> getFulltextSearchResults(SearchResponse searchResponse) {
+  private Optional<FulltextSearchResults> getFulltextSearchResults(SearchResponse searchResponse,
+      boolean enableAutoSuggest,
+      boolean enableBestBets) {
     Optional<ResponseBody> responseBody = searchResponse.get(new ResponseBodyDataExtractionStrategy());
     if (responseBody.isPresent()) {
       Highlighting highlighting = searchResponse.get(new HighlightingDataExtractionStrategy())
           .orElse(FallbackHighlighting.getInstance());
-      return Optional.of(FulltextSearchResults.builder()
-          .totalResultsFound(responseBody.get().getNumFound())
-          .currentResultPage(responseBody.get().getStart())
-          .results(getProcessedResults(responseBody.get().getDocs(), highlighting))
-          .build());
+      Stream<Result> results = getProcessedResults(responseBody.get().getDocs(), highlighting);
+      if (enableBestBets) {
+        log.debug("Best bets is enabled. Results will be sorted so that best bet results are on top.");
+        results = results.sorted(Comparator.comparing(Result::isBestBet).reversed());
+      }
+      FulltextSearchResults.FulltextSearchResultsBuilder fulltextSearchResultsBuilder =
+          FulltextSearchResults.builder()
+              .totalResultsFound(responseBody.get().getNumFound())
+              .currentResultPage(responseBody.get().getStart())
+              .results(results.collect(Collectors.toList()));
+      if (enableAutoSuggest) {
+        log.debug("Auto suggest is enabled.");
+        searchResponse.get(new SuggestionDataExtractionStrategy()).flatMap(suggestion -> LoggedOptional.of(suggestion,
+                logger -> logger.warn("No suggestion has been found in search response")))
```

Comment:
not sure if we should log this on warn level. it is very likely that we don't get a suggestion as users might input any value. can you change to debug level instead?